metricflow/specs/specs.py

```diff
@@ -292,6 +295,67 @@ def accept(self, visitor: InstanceSpecVisitor[VisitorOutputT]) -> VisitorOutputT
         return visitor.visit_dimension_spec(self)
 
 
+class TimeDimensionSpecField(Enum):
+    """Fields of the time dimension spec.
+
+    The value corresponds to the name of the field in the dataclass. This should contain all fields, but implementation
+    is pending.
+    """
+
+    TIME_GRANULARITY = "time_granularity"
+
+
+class TimeDimensionSpecComparisonKey:
+    """A key that can be used for comparing / grouping time dimension specs.
+
+    Useful for assessing if two time dimension specs are equal while ignoring specific attributes. Keys must have the
+    same set of excluded attributes to be valid for comparison.
+
+    This is useful for ambiguous group-by-item resolution where we want to select a time dimension regardless of the
+    grain.
+    """
+
+    def __init__(self, source_spec: TimeDimensionSpec, exclude_fields: Sequence[TimeDimensionSpecField]) -> None:
+        """Initializer.
+
+        Args:
+            source_spec: The spec that this key is based on.
+            exclude_fields: The fields to ignore when determining equality.
+        """
+        self._excluded_fields = frozenset(exclude_fields)
+        self._source_spec = source_spec
+
+        # This is a list of field values of TimeDimensionSpec that we should use for comparison.
+        spec_field_values_for_comparison: List[
+            Union[str, Tuple[EntityReference, ...], TimeGranularity, Optional[DatePart]]
+        ] = [self._source_spec.element_name, self._source_spec.entity_links]
```

Comment:
Would it be worth it to instead allow for a `unique_name and type` style equality comparison for all of these spec classes? I can see this being useful for more than just time dimension specs, although right now they're the only ones that have all of these extra properties.

Reply:
@tlento - mind elaborating? Would those be public properties?

Reply:
They don't have to be public properties. I just mean doing a comparison based on the name/type of spec. Everything else we layer into specs as properties is transient information or something we need to propagate along, like annotations about aggregation state, aliases, stuff like that.

I did just realize that you're using the entire entity links, so my original suggestion doesn't work. That said, our comparison key is currently entity link path and name for linkable specs, and I suspect would be name for everything else. Will we need other properties? If we don't have specific things we'll need to add to that comparison then maybe we just add a method that does the simple thing.